bootstore/src/schemes/v0/peer.rs

```diff
@@ -0,0 +1,810 @@
+                    // though, so we don't panic.
+                    error!(self.log, "Error on connection: {e}");
+                } else {
+                    self.deliver_envelopes().await;
+                }
+            }
+            ConnToMainMsgInner::Disconnected { peer_id } => {
+                // Ignore the stale message if the unique_id doesn't match what
+                // we have stored.
+                if let Some(handle) = self.established_connections.get(&peer_id)
+                {
+                    if unique_id != handle.unique_id {
+                        return;
+                    }
+                }
+                warn!(self.log, "peer disconnected {peer_id}");
+                self.established_connections.remove(&peer_id);
+                self.fsm.on_disconnected(&peer_id);
+            }
+            ConnToMainMsgInner::Received { from, msg } => {
+                match self.fsm.handle_msg(Instant::now().into(), from, msg) {
+                    Ok(None) => self.deliver_envelopes().await,
+                    Ok(Some(api_output)) => {
+                        self.handle_api_output(api_output).await
+                    }
+                    Err(err) => self.handle_api_error(err).await,
+                }
+            }
+            ConnToMainMsgInner::FailedServerHandshake {
+                addr: SocketAddrV6,
+            } => {}
+        }
+    }
+
+    async fn manage_connections(&mut self, peers: BTreeSet<SocketAddrV6>) {
+        if peers == self.peers {
+            return;
+        }
+
+        let peers_to_remove: BTreeSet<_> =
+            self.peers.difference(&peers).cloned().collect();
+        let new_peers: BTreeSet<_> =
+            peers.difference(&self.peers).cloned().collect();
+
+        self.peers = peers;
+
+        // Start a new client for each peer that has an addr < self.config.addr
+        for addr in new_peers {
+            if addr < self.config.addr {
+                self.handle_unique_id_counter += 1;
+                let handle = spawn_connection_initiator_task(
+                    self.handle_unique_id_counter,
+                    self.config.id.clone(),
+                    self.config.addr.clone(),
+                    addr.clone(),
+                    &self.log,
+                    self.conn_tx.clone(),
+                )
+                .await;
+                self.initiating_connections.insert(addr, handle);
+            }
+        }
+
+        // Remove each peer that we no longer need a connection to
+        for addr in peers_to_remove {
+            self.remove_peer(addr).await;
+        }
+    }
+
+    async fn remove_accepted_connection(&mut self, addr: &SocketAddrV6) {
+        if let Some(handle) = self.accepted_connections.remove(&addr) {
+            // The connection has not yet completed its handshake
+            let _ = handle.tx.send(MainToConnMsg::Close).await;
+        }
+    }
```

Comment:
Worth adding a warning here if a connection isn't found in the `accepted_connections` map.

Reply:
Similar rationale to `established_connections` above